test/core/list/each.wren

```diff
@@ -0,0 +1,3 @@
+var words = ""
+["One", "Two", "Three"].each {|word| words = words + word }
+IO.print(words) // expect: OneTwoThree
```

Comment:
This is swell. Can you also test:

1. Passing an empty list to it.
2. Passing something that doesn't have a `.call` method.

I know it seems pointless for something so simple, but good test coverage matters a lot.